File: code-samples/analytics-old/quick-start.js
const RC_SDK = require('@ringcentral/sdk').SDK
require('dotenv').config();

var rcsdk = new RC_SDK({
    'server':       process.env.RC_SERVER_URL,
    'clientId':     process.env.RC_CLIENT_ID,
    'clientSecret': process.env.RC_CLIENT_SECRET
});
var platform = rcsdk.platform();
platform.login({
	  'jwt':  process.env.RC_JWT
})

let FROM_DATE = '2022-04-12T07:00:00.000Z'
let TO_DATE   = '2022-05-11T07:00:00.000Z'

platform.on(platform.events.loginSuccess, function(e){
    run_report( FROM_DATE, TO_DATE )
});

async function run_report( from_time, to_time ) {
	try {
		let options = {
				"grouping":{
					"groupBy":"Users"
				},
				"timeSettings":{
					"timeZone": "US/Pacific",
					"timeRange":{
						"timeFrom": from_time,
						"timeTo": to_time
			}
				},
				"responseOptions":{
					"counters":{
						"allCalls":{
							"aggregationType":"Sum"
					}
			}
				}
		}
		let result = await platform.post("/analytics/calls/v1/accounts/~/aggregation/fetch", options);
		let response = await result.json();
		console.dir(response, {depth: null});
	} catch (e) {
			console.log(e.message);
	}
}

File: code-samples/analytics-old/quick-start.rb
require 'ringcentral'
require 'dotenv/load'

FROM_DATE = '2022-04-12T07:00:00.000Z'
TO_DATE   = '2022-05-11T07:00:00.000Z'

$rc = RingCentral.new( ENV['RC_CLIENT_ID'],
                       ENV['RC_CLIENT_SECRET'],
                       ENV['RC_SERVER_URL'] )

$rc.authorize(jwt: ENV['RC_JWT'])       

def run_report(from_date, to_date)
  resp = $rc.post('/analytics/calls/v1/accounts/~/aggregation/fetch', payload: {
    "grouping":{ 
      "groupBy": "Users" 
    },
    "timeSettings":{
      "timeZone": "US/Pacific",
	    "timeRange":{
	      "timeFrom": from_date,
	      "timeTo": to_date
	    }
    },
    "responseOptions":{
	    "counters":{ 
        "allCalls":{ 
          "aggregationType": "Sum" 
        } 
      }
    }
  })
  puts "Response: " + JSON.pretty_generate(resp.body)
end

run_report( FROM_DATE, TO_DATE )
